src/main/java/net/fabricmc/fabric/mixin/registry/MixinIdRegistry.java

```diff
@@ -79,87 +79,120 @@ public void setPre(int id, Identifier identifier, Object object, CallbackInfoRet
 	@Inject(method = "set", at = @At("RETURN"))
 	public void setPost(int id, Identifier identifier, Object object, CallbackInfoReturnable info) {
 		SimpleRegistry<Object> registry = (SimpleRegistry<Object>) (Object) this;
-		if (listeners != null) {
-			for (RegistryListener listener : listeners) {
+		if (fabric_listeners != null) {
+			for (RegistryListener listener : fabric_listeners) {
 				listener.afterRegistryRegistration(registry, id, identifier, object);
 			}
 		}
 	}
 
 	@Override
-	public void remap(Object2IntMap<Identifier> idMap, boolean reallocateMissingEntries) throws RemapException {
+	public void remap(Object2IntMap<Identifier> remoteIndexedEntries, RemapMode mode) throws RemapException {
 		//noinspection unchecked, ConstantConditions
 		SimpleRegistry<Object> registry = (SimpleRegistry<Object>) (Object) this;
 
-		Object defaultValue = null;
-		//noinspection ConstantConditions
-		if (registry instanceof DefaultedRegistry) {
-			defaultValue = registry.get(((DefaultedRegistry<Object>) registry).getDefaultId());
-		}
-
-		if (!reallocateMissingEntries && !idMap.keySet().equals(registry.getIds())) {
-			throw new RemapException("Source and destination keys differ!");
+		// Throw on invalid conditions.
```

Comment:
This should be a javadoc on the encompassing method, describing the behaviour

Reply:
This is *super* internal code. If you're an user looking at it, you've done many things wrong.